### PR TITLE
hide tournament button when there is no tournaments

### PR DESCRIPTION
**Pull request by smallst**

### CODE REVIEW

File: app/views/courses/CoursesView.coffee [hunk 1]
```diff
@@ -71,6 +72,16 @@ module.exports = class CoursesView extends RootView
     @originalLevelMap = {}
     @urls = require('core/urls')
 
+    @hasActiveTournaments = false
+    tournaments = new CocoCollection([], { url: "/db/tournaments?memberId=#{me.id}", model: Tournament})
+    @listenToOnce tournaments, 'sync', =>
+      tournamentsByClass = (t.toJSON() for t in tournaments.models)[0]
```
Comment (AndrewJakubowicz) [inline app/views/courses/CoursesView.coffee:78]: I don't quite follow this line. Is it possible for there to be an error here, or is there always a 0 index?

**Review by AndrewJakubowicz — APPROVED** [review]
I manually tested a student with proxy that isn't in a tournament and it looks good.

Left a tiny nitpick as I don't quite understand the structure of the tournaments collection.
Thank you!

File: app/views/courses/CoursesView.coffee (hunk 1 above)
Comment (smallst) [reply app/views/courses/CoursesView.coffee:78]: it at least shapes as `[{}]` so it always contains 0 index :joy: and all lodash functions won't cause error with empty object/array

**Comment by smallst** [comment]
@AndrewJakubowicz sorry for the stange structure of tournament collection . for a students, that url returns an object like this:
```{
 class1: [ tournaments1, tournament2 .. ],
 class2: [ tournaments1, tournament2...]
...
}```
so i need to unpack it. 
this structure previously works only for the `mainTournament` page, so i arrange it as a complex josn. and this time i don't want to make server code changes so i reuse this weird structure :joy:

File: app/views/courses/CoursesView.coffee (hunk 1 above)
Comment (AndrewJakubowicz) [reply app/views/courses/CoursesView.coffee:78]: Looks good to me!

**Pull request merged**